### DISCUSSION TITLE
How to reload a component?

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [elimintz/justpy#440] [Q&A]
Thank you for this excellent library.

I am using aggrid to display a table. I'd love to make sure the content is refreshed on a click of a button.

the following reload obviously does **not do this** but adds another aggrid ... **What needs to be changed to replace the aggrid with a "new" aggrid from a pandas dataframe or reuse the existing aggrid for the purpose?**

```python
def reload(self,_msg=None):
        '''
        reload clicked
        '''
        grid = self.df.jp.ag_grid(a=self.container)
        grid.row_data_div = self.row_data_div
        grid.on('rowSelected', self.row_selected)
        grid.options.columnDefs[0].checkboxSelection = True
```

**Comment by sandeep-gh** [comment]
I haven't used ag_grid. Why are you creating a new grid on reload? Just update the data for the current grid. It will get refreshed.
